### PR TITLE
#3886: fixed error in corrmap with labels_ eq None

### PR DESCRIPTION
**Pull request by ppasler**
#3886: fixed error in corrmap with labels_ eq None

### CODE REVIEW

**Comment by ppasler** [comment]
I would extract the labels check to a method? It is used 5 times inside the class?

    def labelsPresent(ica):
        return (not hasattr(ica, 'labels_') or ica.labels_ is None)

**Comment by jaeilepp** [comment]
I think the right way to do this is to make sure that ICA always has the `labels_` attribute (i.e. set it to `None` at the ICA constructor).
@ppasler  are you up for it?

**Comment by jaeilepp** [comment]
Actually, I thought the problem was something else, but my point still stands. Maybe the ICA constructor should set the ``self.labels_ = dict()`` before populating the labels, so the checks here would be unnecessary. Makes sense?

**Comment by jona-sassenhagen** [comment]
Yes, I think so too. I didn't do it originally because I didn't want to mess with the class, and @Eric89GXL implemented the `labels_` field.

Setting it up so that ICA comes with an empty labels_ dict would be good.

**Comment by ppasler** [comment]
But the issue persist, if `ica_misc['labels_']` in `read_ica()` is None.
So i have to check if `ica_misc['labels_']` is not None [ica.py#L1921](https://github.com/mne-tools/mne-python/blob/master/mne/preprocessing/ica.py#L1921)

**Comment by jaeilepp** [comment]
Yeah, that's true. But then we can get rid of the ``hasattr`` mess. So the check would be simply
```
if label is not None and ica.labels_ is None:
```

**Comment by jaeilepp** [comment]
No wait, I'm wrong again. There should be a check in ``read_ica``, so it does not set it to None.

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -2413,7 +2411,7 @@ def corrmap(icas, template, threshold="auto", label=None, ch_type="eeg",
         logger.info('Displaying selected ICs per subject.')
 
     for ii, (ica, max_corr) in enumerate(zip(icas, mx)):
-        if (label is not None) and (not hasattr(ica, 'labels_')):
+        if label is not None:
```
Comment (jona-sassenhagen) [inline mne/preprocessing/ica.py]: Uh oh ... This will overwrite existing labels.

Comment (ppasler) [inline mne/preprocessing/ica.py]: You're right, my bad. Do we need this condition anyway? `ica.labels_` is always set in this branch, so no need to set it to an empty `dict`.

Comment (jaeilepp) [inline mne/preprocessing/ica.py]: Yeah, I think this check can be removed.

**Comment by ppasler** [comment]
Could somebody restart travis-ci please, it seems the ftp server wasn't available?

**Comment by ppasler** [comment]
@jona-sassenhagen @jaeilepp could you please approve and merge this?

**Comment by jona-sassenhagen** [comment]
It looks good to me. Do the docs need change (document that ICA has an attr "labels_" now)?

**Comment by jaeilepp** [comment]
> Do the docs need change (document that ICA has an attr "labels_" now)?

There are already a couple of lines at the [ica tutorial](http://mne-tools.github.io/dev/auto_tutorials/plot_artifacts_correction_ica.html).

**Comment by jaeilepp** [comment]
Thanks @ppasler

**Comment by jona-sassenhagen** [comment]
Thanks @ppasler !

**Comment by ppasler** [comment]
No thanks needed, I also profit from this fix :)

**Pull request merged**